internal/db/db_blueprints.go

```diff
@@ -27,6 +51,44 @@ const (
 	sqlDeleteBlueprint = `DELETE FROM blueprints WHERE id = $1 AND org_id = $2 AND account_number = $3`
 )
 
+func (db *dB) GetBlueprintComposes(orgId string, blueprintId uuid.UUID, since time.Duration, limit, offset int, ignoreImageTypes []string) ([]ComposeEntryWithVersion, int, error) {
+	ctx := context.Background()
+	conn, err := db.Pool.Acquire(ctx)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer conn.Release()
+	result, err := conn.Query(ctx, sqlGetBlueprintComposes, orgId, blueprintId, since, ignoreImageTypes, limit, offset)
+	defer result.Close()
+
+	if err != nil {
+		return nil, 0, err
+	}
+
+	var composes []ComposeEntryWithVersion
+	for result.Next() {
+		entry := ComposeEntryWithVersion{
+			BlueprintId: blueprintId,
+		}
+		err = result.Scan(&entry.BlueprintVersion, &entry.Id, &entry.Request, &entry.CreatedAt, &entry.ImageName, &entry.ClientId)
+		if err != nil {
+			return nil, 0, err
+		}
+		composes = append(composes, entry)
+	}
+	if err = result.Err(); err != nil {
+		return nil, 0, err
+	}
+
+	var count int
+	err = conn.QueryRow(ctx, sqlCountActiveBlueprintComposesSince, orgId, blueprintId, since, ignoreImageTypes).Scan(&count)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return composes, count, nil
+}
```

Comment:
This function is responsible for two things, list and count. I suggest to split it into two db functions which can be called separately and possibly reused, there is no performance benefit in doing it this way.